StackExchange/Stacks
# test(a11y): switch to custom APCA axe color contrast checks

This PR adjusts our a11y checks to aim for [APCA bronze conformance level](https://readtech.org/ARC/tests/visual-readability-contrast/?tn=criterion) instead of WCAG 2.1 AA. This is in preparation of the Stacks Color Palette Redefinition work we are working on in [this branch](https://github.com/StackExchange/Stacks/tree/stacks-next).

Here is what has been done in detail:
- Create a "axe-apca" (pseudo-package) containing 2 axe-core custom rules: `color-contrast-apca-bronze` and `color-contrast-apca-silver`
- Register the newly created `color-contrast-apca-bronze` rule in our `test-utils` file responsible for generating a11y tests for our components
- For `light` and `dark` mode, switch off the WCAG 2.1 AA axe-core rule and use `color-contrast-apca-bronze` instead
- For `highcontrast` mode, activate the WCAG 2.1 AAA axe-core rule (which was not turned on yet)

As part of this PR many a11y test cases have been marked to be skipped because they are failing APCA bronze conformance level. This is a tradeoff we are happy to temporary live with since we expect to reactivate most (if not all) of those tests once the new color palette is in place.  

> **Note**
> For the time being we are keeping the axe-apca package directly within this repository but it should be handled so that we feel comfortable extracting it out at any point (this is why the axe-apca code is currently in its own folder with its own set of unit tests, a readme and a pseudo package.json). As we roll out our new color palette in internal codebases this package should be extracted so that we could register the new  `color-contrast-apca-bronze` in their respective automated tests. Furthermore the axe-core APCA rules are pioneering work which would benefit from being open sourced and getting feedback/contribution from a wider community.


## How to test

- Run `npm run test:a11y` and `npm run test:unit` both command should still pass as expected
- Change the color or background style for one of the components we have a11y tests for and observe the test failing (e.g. change line 13 of `activity-indicator.less` to `var(--red-100)` and rerun `npm run test:a11y`)
